src/jttt/Core/AlphaBetaStrategy.java

```diff
@@ -0,0 +1,112 @@
+    }
+
+    public int calculateNextMove(Board board, Counter counter) {
+        this.counter = counter;
+        DEPTH = board.numberOfOpenPositions();
+        return indexToDisplayPosition(aBMinimax(DEPTH, this.counter, INITIAL_ALPHA, INITIAL_BETA, board));
+    }
+
+    private HashMap<String, Integer> aBMinimax(int depth, Counter currentCounter, int alpha, int beta, Board currentBoard) {
+        Integer bestScore = setInitialBestScore(currentCounter);
+        Integer bestMove = -1;
+        if (noMoreMovesAvailable(depth, currentBoard)) {
+            return calculateResult(currentBoard, depth);
+        }
+
+        for (Integer move : currentBoard.remainingPositions()) {
+            Board currentStateOfBoard = currentBoard.newBoardWithNewMove(move, currentCounter);
+            Integer score = aBMinimax(  depth - 1,
+                                        currentCounter.opponentCounter(),
+                                        alpha,
+                                        beta,
+                                        currentStateOfBoard).get(SCORE_KEY);
+            if (newBestScore(currentCounter, bestScore, score)) {
+                bestScore = score;
+                bestMove = move;
+            }
+            alpha = newAlpha(alpha, score, currentCounter);
+            beta = newBeta(beta, score, currentCounter);
+            if (beta <= alpha) {
+                break;
+            }
+        }
+        return createResultMap(bestScore, bestMove);
+    }
+
+    private int newAlpha(int alpha, Integer score, Counter currentCounter) {
+        if (isComputerCounter(currentCounter)) {
+            return max(alpha, score);
+        }
+        return alpha;
+    }
+
+    private int newBeta(int beta, Integer score, Counter currentCounter) {
+        if (!isComputerCounter(currentCounter)) {
+            return min(beta, score);
+        }
+        return beta;
+    }
+
+    private int indexToDisplayPosition(HashMap<String, Integer> result) {
+        return result.get(MOVE_KEY) + DISPLAY_POSITION_OFFSET;
+    }
+
+    private boolean newBestScore(Counter currentCounter, Integer bestScore, Integer score) {
+        return isComputerCounter(currentCounter) && score > bestScore ||
+                !isComputerCounter(currentCounter) && score < bestScore;
+    }
+
+    private boolean isComputerCounter(Counter currentCounter) {
+        return this.counter == currentCounter;
+    }
+
+    private boolean noMoreMovesAvailable(int depth, Board currentBoard) {
+        return currentBoard.isGameOver() || depth == 0;
+    }
+
+    private Integer setInitialBestScore(Counter currentCounter) {
+        return currentCounter == this.counter ? -INITIAL_SCORE : INITIAL_SCORE;
+    }
+
+    private HashMap<String, Integer> calculateResult(Board currentBoard, int depth) {
+        double score = INITIAL_SCORE / (currentBoard.boardSize() - currentBoard.numberOfOpenPositions());
+        if (currentBoard.findWinner().equals(Counter.EMPTY)) {
```

Comment:
this conditional could be extract to something like `if( hasWinner(currentBoard)) {`